src/components/structures/auth/Registration.js

```diff
@@ -69,26 +69,6 @@ module.exports = React.createClass({
     getInitialState: function() {
         const serverType = ServerType.getTypeFromHsUrl(this.props.customHsUrl);
 
-        const customURLsAllowed = !SdkConfig.get()['disable_custom_urls'];
-        let initialPhase = this.getDefaultPhaseForServerType(serverType);
-        if (
-            // if we have these two, skip to the good bit
-            // (they could come in from the URL params in a
-            // registration email link)
-            (this.props.clientSecret && this.props.sessionId) ||
-            // if custom URLs aren't allowed, skip to form
-            !customURLsAllowed ||
-            // if other logic says to, skip to form
-            this.props.skipServerDetails
```

Comment:
You should also remove `skipServerDetails` and the functions supplied to it in `MatrixChat`.